src/utils/setPermissions.js

```diff
@@ -0,0 +1,6 @@
+function setPermissions(controllerClass, permissions) {
```

Comment:
this is not a standard form of a decorator

Decorator should be a curried function, something like a 
```
const permissionValidationDecorator = permissionRules => TargetClass =>
  class PermissionValidationWrapper extends TargetClass {
    permissionRules = permissionRules;
  };

export default permissionValidationDecorator;
```

This will allow further functional composition